File: 2a.pl
#!/usr/bin/perl
use warnings;
use strict;
use feature qw{ say };

use ARGV::OrDATA;

my ($depth, $horiz) = (0, 0);
while (<>) {
    my ($direction, $amount) = split;
    {
        forward => sub { $horiz += $amount },
        up      => sub { $depth -= $amount },
        down    => sub { $depth += $amount }
    }->{$direction}();
}
say $depth * $horiz;

__DATA__
forward 5
down 5
forward 8
up 3
down 8
forward 2

File: 2a.py
#! /usr/bin/python3
import sys

depth = 0
horiz = 0

def forward(amount):
    global horiz
    horiz += amount

def up(amount):
    global depth
    depth -= amount

def down(amount):
    global depth
    depth += amount

with open(sys.argv[1], 'r') as fh:
    for line in fh:
        (direction, amount) = line.split()
        {
            'forward': forward,
            'up'     : up,
            'down'   : down
        }[direction](int(amount))
print(depth * horiz)
